roxie/ccd/ccdserver.cpp

```diff
@@ -4557,9 +4561,38 @@ class CRemoteResultAdaptor : implements IEngineRowStream, implements IFinalRoxie
             //       But this could still cause too many reply packets on the fastlane
             //       (higher priority output Q), which may cause the activities on the 
             //       low priority output Q to not get service on time.
+
+            int origPriority = (int)ctx->queryOptions().priority;
+            int dynPriority = ctx->queryOptions().dynPriority;
+            if (dynPriority < origPriority)
+            {
+                unsigned newPri = ROXIE_SLA_PRIORITY + ROXIE_HIGH_PRIORITY;
+                switch (dynPriority)
+                {
+                    case 1:
+                        newPri = ROXIE_HIGH_PRIORITY;
+                        break;
+                    case 0:
+                        newPri = ROXIE_LOW_PRIORITY;
+                        break;
+                }
+                p->queryHeader().activityId &= ~ROXIE_PRIORITY_MASK;
+                p->queryHeader().activityId |= newPri;
+            }
+
+            // TODO: perhaps check elapsed every Nth msg ?
+            if ( (dynPriorityAdjustTime > 0) && (origPriority == 0) && (dynPriority == 0) &&
+                 (ctx->queryElapsedMs() > dynPriorityAdjustTime) )
```

Comment:
This may have a notable impact, is it possible to measure?  Slightly better would be to convert the time to cycles and avoid the cycle->ms conversion on each test (avoids a floating point multiply).  May be insignificant.